.github/workflows/maven.yml

```diff
@@ -0,0 +1,37 @@
+name: maven
+
+on:
+  push:
+    branches:
+      - master
+  pull_request:
+    types: [opened, labeled, unlabeled, synchronize]
+
+permissions:
+  contents: read
+
+env:
+  GRADLE_OPTS: "-Dorg.gradle.jvmargs=-Xmx4g -Dorg.gradle.daemon=false -Dkotlin.incremental=false"
+
+jobs:
+  test_maven:
+    permissions:
+      checks: write # for actions/upload-artifact
+    runs-on: ubuntu-latest
+    if: github.ref == 'refs/heads/master' || contains(github.event.pull_request.labels.*.name, 'maven')
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
```

Comment:
This is actually slower than the default: https://github.com/actions/checkout?tab=readme-ov-file#fetch-all-history-for-all-tags-and-branches

Reply:
https://github.com/square/okhttp/pull/8929